pkg/auth/rhsso_authenticator.go

```diff
@@ -305,14 +301,18 @@ func (a *RHSSOAuthenticator) CreateAuthenticator() func(name, in string, authent
 			}
 			//this code is part of the authorization process and should move to authz_handler
 			//after https://github.com/go-openapi/runtime/issues/158 is resolved
-			clusterID := params.GetParam(r.Context(), params.ClusterId)
-			ownedBy, err := a.isClusterOwnedByUser(clusterID, p.(*ocm.AuthPayload))
+			ownedBy := true
+			if clusterID := params.GetParam(r.Context(), params.ClusterId); clusterID != "" {
+				ownedBy, err = a.isObjectOwnedByUser(clusterID, &common.Cluster{}, p.(*ocm.AuthPayload))
+			} else if infraEnvID := params.GetParam(r.Context(), params.InfraEnvId); infraEnvID != "" {
+				ownedBy, err = a.isObjectOwnedByUser(infraEnvID, &common.InfraEnv{}, p.(*ocm.AuthPayload))
+			}
 			if err != nil {
-				log.Errorf("Fail to verify access to cluster. Error %v", err)
+				log.Errorf("Faied to verify access to object. Error %v", err)
 				return true, nil, common.NewApiError(http.StatusInternalServerError, err)
 			}
 			if !ownedBy {
-				return true, nil, common.NewApiError(http.StatusNotFound, errors.New("Cluster Not Found"))
+				return true, nil, common.NewApiError(http.StatusNotFound, errors.New("Object Not Found"))
```

Comment:
it would be helpful if the error said exactly which object was causing the issue, i.e: "Cluster <id> Not Found", or "Infra-Env <id> not Found"